src/components/PeopleFilters.tsx

```diff
@@ -1,23 +1,95 @@
+import { Link, useLocation, useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+import classNames from 'classnames';
+import React from 'react';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const location = useLocation();
+
+  function handleCenturyClick(century: string) {
+    const currentCenturies = searchParams.getAll('centuries');
+    let newCenturies: string[] = [];
+
+    if (currentCenturies.includes(century)) {
+      newCenturies = currentCenturies.filter(cent => cent !== century);
+    } else {
+      newCenturies = [...currentCenturies, century];
+    }
+
+    return getSearchWith(searchParams, { centuries: newCenturies });
+  }
+
+  function isActive(paramKey: string, value: string) {
+    const paramValues = searchParams.getAll(paramKey);
+
+    return paramValues.includes(value);
+  }
+
+  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const newParams = new URLSearchParams(searchParams);
+
+    const queryValue = event.target.value;
+
+    if (queryValue) {
+      newParams.set('query', queryValue);
+    } else {
+      newParams.delete('query');
+    }
+
+    setSearchParams(newParams);
+  }
+
+  function resetFilters() {
+    const newSearch = new URLSearchParams(searchParams.toString());
+
+    newSearch.delete('sex');
+    newSearch.delete('query');
+    newSearch.delete('centuries');
+
+    return newSearch.toString();
```

Comment:
The `resetFilters` function should use `setSearchParams` to update the URL with the new search parameters. Currently, it returns a string, but it should instead call `setSearchParams(newSearch)` to apply the changes.